SharpMUSH.Implementation/Functions/MailFunctions.cs

```diff
@@ -1,64 +1,611 @@
-﻿using SharpMUSH.Library.Attributes;
+﻿using SharpMUSH.Implementation.Commands.MailCommand;
+using SharpMUSH.Library;
+using SharpMUSH.Library.Attributes;
 using SharpMUSH.Library.Definitions;
+using SharpMUSH.Library.DiscriminatedUnions;
+using SharpMUSH.Library.Extensions;
+using SharpMUSH.Library.Models;
 using SharpMUSH.Library.ParserInterfaces;
+using SharpMUSH.Library.Queries.Database;
+using SharpMUSH.Library.Services.Interfaces;
 
 namespace SharpMUSH.Implementation.Functions;
 
 public partial class Functions
 {
+	/// <summary>
+	/// Helper to parse folder:message format and retrieve mail
+	/// </summary>
+	private static async ValueTask<(string folder, int messageIndex, SharpMail? mail)> GetMailMessage(
```

Comment:
- Most of the functions above already split up folder and mail. So why take in messageSpec here?